site/src/modules/resources/AgentLogLine.tsx

```diff
@@ -0,0 +1,60 @@
+import type { Interpolation, Theme } from "@emotion/react";
+import AnsiToHTML from "ansi-to-html";
+import { type FC, type ReactNode, useMemo } from "react";
+import {
+  type Line,
+  LogLine,
+  LogLinePrefix,
+  LogLineSpace,
+} from "components/Logs/LogLine";
+
+const convert = new AnsiToHTML();
+
+interface AgentLogLineProps {
+  line: Line;
+  number: number;
+  style: React.CSSProperties;
+  sourceIcon: ReactNode;
+  maxNumber: number;
+}
+
+export const AgentLogLine: FC<AgentLogLineProps> = ({
+  line,
+  number,
+  maxNumber,
+  sourceIcon,
+  style,
+}) => {
+  const output = useMemo(() => {
+    return convert.toHtml(line.output.split(/\r/g).pop() as string);
+  }, [line.output]);
+
+  return (
+    <LogLine css={{ paddingLeft: 16 }} level={line.level} style={style}>
+      {sourceIcon}
+      <LogLinePrefix
+        css={styles.number}
+        style={{
+          minWidth: `${maxNumber ? maxNumber.toString().length - 1 : 0}em`,
```

Comment:
nit: `maxNumber` will only be falsey when it's `0`, and `0..toString().length - 1` is `0`, so the condition is a bit redundant.

it'd make more sense if `maxNumber` could be `undefined`, but that's not what the type says.